kafka-client-lib/src/main/java/io/jenkins/plugins/remotingkafka/KafkaProducerClient.java

```diff
@@ -27,7 +29,11 @@ public static KafkaProducerClient getInstance() {
     public Producer<String, byte[]> getByteProducer(Properties producerProps) {
         if (byteProducer == null) {
             synchronized (KafkaProducerClient.class) {
-                if (byteProducer == null) byteProducer = new KafkaProducer<>(producerProps);
+                if (byteProducer == null) {
+                    Thread.currentThread().setContextClassLoader(
```

Comment:
You do not restore it after the call, so it may impact the calling threads. Is it as designed? I doubt